composer/utils/dist.py

```diff
@@ -74,7 +74,6 @@ def get_local_rank() -> int:
         int: The local world size
     """
     local_rank = _get_distributed_config_var(env_var="LOCAL_RANK", human_name="local rank", default=0)
-    assert local_rank == get_global_rank() % get_local_world_size(), "invariant violation"
```

Comment:
This change is unrelated to the main purpose of this PR, but I didn't feel like creating a new PR for it. Essentially, we shouldn't be assuming that all nodes have the same amount of processes.

Reply:
If we are removing this assumption, we should also add in a `NODE_RANK` env variable, and update `get_node_rank` to use that.

Reply:
I'm game.